src/java.security.jgss/share/classes/sun/security/jgss/LoginConfigImpl.java

```diff
@@ -165,30 +175,42 @@ public Configuration run() {
      * the system-wide Configuration object.
      */
     private AppConfigurationEntry[] getDefaultConfigurationEntry() {
-        HashMap <String, String> options = new HashMap <String, String> (2);
+        HashMap <String, String> gssOptions = new HashMap <String, String> (2);
+        HashMap <String, String> krb5Options = new HashMap <String, String> (2);
 
-        if (mechName == null || mechName.equals("krb5")) {
+        if (mechName == null || mechName.equals("krb5") || useNative) {
             if (isServerSide(caller)) {
+                gssOptions.put("useDefaultCreds", "true");
+                gssOptions.put("doNotPrompt", "true");
+                gssOptions.put("accept", "true");
                 // Assuming the keytab file can be found through
                 // krb5 config file or under user home directory
-                options.put("useKeyTab", "true");
-                options.put("storeKey", "true");
-                options.put("doNotPrompt", "true");
-                options.put("principal", "*");
-                options.put("isInitiator", "false");
+                krb5Options.put("useKeyTab", "true");
+                krb5Options.put("storeKey", "true");
+                krb5Options.put("doNotPrompt", "true");
+                krb5Options.put("principal", "*");
+                krb5Options.put("isInitiator", "false");
             } else {
                 if (caller instanceof HttpCaller && !HTTP_USE_GLOBAL_CREDS) {
-                    options.put("useTicketCache", "false");
+                    gssOptions.put("tryDefaultCreds", "false");
+                    krb5Options.put("useTicketCache", "false");
                 } else {
-                    options.put("useTicketCache", "true");
+                    gssOptions.put("tryDefaultCreds", "true");
+                    krb5Options.put("useTicketCache", "true");
                 }
-                options.put("doNotPrompt", "false");
+                gssOptions.put("initiate", "true");
+                gssOptions.put("doNotPrompt", "false");
+                krb5Options.put("doNotPrompt", "false");
             }
             return new AppConfigurationEntry[] {
                 new AppConfigurationEntry(
-                        "com.sun.security.auth.module.Krb5LoginModule",
+                        "com.sun.security.auth.module.GssLoginModule",
```

Comment:
This is a change in behavior, why?

Reply:
Yes, it is.  Whether this is good or not will be up to Oracle.  `GssLoginModule` isn't quite on par with `Krb5LoginModule`'s functionality -- we'd have to add support for using the new `gss_acquire_cred_from()` functions in MIT and Heimdal in order to add support for the missing functionality, which is all about acquiring credentials with raw passwords or raw keys.

Reply:
That's fine then!